### ISSUE TITLE
✍️ Contribution period: <Luis_Camacho> 

### ISSUE THREAD

**Issue by luiscamachocaballero** [ersilia-os/ersilia#872]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [X] Install the Ersilia Model Hub and test the simplest model
- [X] Write a motivation statement to work at Ersilia
- [x] Submit your first contribution to the Outreachy site

### Week 2 - Install and run an ML model

- [x] Select a model from the suggested list
- [x] Install the model in your system
- [x] Run predictions for the EML
- [x] Compare results with the Ersilia Model Hub implementation!
- [x] Install and run Docker!

### Week 3 - Propose new models

- [x] Suggest a new model and document it (1)
- [x] Suggest a new model and document it (2)
- [x] Suggest a new model and document it (3)

### Week 4 - Prepare your final application

- [x] Submit the final application in the Outreachy website

**Comment by luiscamachocaballero** [comment]
Following the [instructions](https://ersilia.gitbook.io/ersilia-book/ersilia-model-hub/installation), the Ersilia Model Hub installation was very [straightforward.]. Only I suggest editing Pre-requisite 5: The Isaura data lake where it wrongly was put the instruction `conda activate ersilia`. 
Also, the link to install Docker moves to a piece of broad information, and I think it could cause confusion; people could figure out they have to install only Docker Desktop, I mean, Pre-requisite 6 must indicate to install Docker Engine

**Comment by luiscamachocaballero** [comment]
I was a [research](https://orcid.org/0000-0001-6569-550X) manager at [PUCP](https://www.timeshighereducation.com/world-university-rankings/pontifical-catholic-university-peru). In 1998, I founded the Rural Telecommunications Research Group ([GTR](http://gtr.telecom.pucp.edu.pe/)) at PUCP and then I chose the hardest option:  intense field labor in the more extreme rural zones of the world instead of a traditional tenure-track on PUCP campus, I spent 15 years carrying out ICT4D projects in the Amazon rainforest for the sake of low-income small villagers. I started working immediately on connectivity for very isolated rural villages located in the Amazon rainforest. Despite the fact I was a scholar, my team acted as activist practitioners, more focused on the sake of population than on research, [the work was extremely intense](https://get.google.com/albumarchive/102999545716122098189), our major concern was that public health workers had connectivity as a tool to deal with emergencies and fight against neglected diseases like malaria. After many attempts and several failures (we started trying unstable AX.25 packet radio technology), the effort was rewarded, in 2006 we managed to install a chain of WiFi repeaters, and the following year we managed to expand the chain to reach 500 km in length, [from Cabo Pantoja to Iquitos](https://www.google.com/maps/dir/Cabo+Pantoja/Iquitos/@-2.5759911,-74.8111897,8z/data=!4m14!4m13!1m5!1m1!1s0x91d945e66577db4b:0x28d834b07514003e!2m2!1d-75.1715769!2d-0.9701686!1m5!1m1!1s0x91ea10b0e440c1cb:0x9210b8d7040a6692!2m2!1d-73.2516326!2d-3.7436735!3e0!5m1!1e4), we called it [NapoNet](http://gtr.telecom.pucp.edu.pe/download/gtr-medios/D-EC-11112012%20-%20El%20Comercio%20-%20Pa%C3%ADs%20-%20pag%2018-1.pdf), as far I know it's the longest chain in the world. I worked on that for four more years until 2011. I'm happy that 14 years after the installation, the chain is still working, it survived without us!; well, not exactly, we never got away completely, but the responsibility of the maintenance fell to the local authorities. The lesson was clear: long-term support for projects to succeed. In 2013 we started a new project, [Tucan3G](https://www.youtube.com/watch?v=PCcS3dqc3k4&t=20s), which is very similar to Mexican [Rizhomatica](https://www.rhizomatica.org/) but with NapoNet as backhaul and with TELCO Movistar (BME: TEF) as a partner to avoid legal issues for using mobile frequency bands.

In 2016 I started [another big project](https://www.linuxfoundation.org/blog/2016/09/2016-lift-scholarship-winner-luis-camacho-caballero-preserving-amazon-languages-with-linux/): I launched [SIMINCHIKKUNARAYKU](https://www.siminchikkunarayku.pe/) Initiative which proposes a holistic vision: use of artificial intelligence, multimedia, and public policy (based on economical evidence) to preserve and foster South American native languages. Our first target was Quechua, including all its dialects, the most spoken American native language. Nowadays, Artificial Intelligence is fueling a new industrial revolution, Artificial Intelligence also allows the computational portability of languages, and computational portability involves creating systems of natural language processing (NLP). Artificial Intelligence is the disruptive and ultimate technology for the preservation of endangered languages, but it’s clear that alleviating the lack of resources is the first step and that needs big bucks and that funding is not available unfortunately, so I am forced to leave my project stand-by until I reach to convince big donors, angel investors or South American decision-makers make a move in the right direction.

Running my endeavor, I developed new skills like Artificial Intelligence, Natural Language Processing, Big Data, DevOps, and Cloud Computing. I dream big. I am happy I found Outreachy and Ersilia, I feel I can use my skills to make a contribution again for the sake of rural communities threatened by neglected deadly diseases.

**Comment by luiscamachocaballero** [comment]
Dealing with COVID is urgent but I think many people are already working on that. So, that is why I selected Plasma Protein Binding (IDL-PPBopt). As far as I know, PPB means that if there is no protein that transports some kind of hormone, that hormone does not work. So a non-empirical treatment alternative is that there is a substance or a medication that increases the binding of a protein to a hormone or that causes this protein to increase in quantity in the blood so that, perhaps, the hormones, whose production is deficient, can join together in sufficient numbers to be transported to the organ that needs the hormone.
For example, if there is no binding protein then it happens a disease known as hypothyroidism or deficiency of adequate amounts of thyroid hormone. On the other hand, the other extreme is also negative, sometimes proteins exist in such quantities that they bind closely to the hormone or chemical substance, causing damage. Both excess and deficiency of hormones are harmful.

**Comment by luiscamachocaballero** [comment]
`(ersilia) alonso@alonso-OMEN-by-HP-Laptop:~/ersilia$ ersilia serve eos22io
🚀 Serving model eos22io: idl-ppbopt

   URL: http://127.0.0.1:47081
   PID: 545296
   SRV: conda

👉 To run model:
   - run

💁 Information:
   - info
(ersilia) alonso@alonso-OMEN-by-HP-Laptop:~/ersilia$ ersilia api -i 'CCCOCCC'
{
    "input": {
        "key": "POLCUAVZOMRGSN-UHFFFAOYSA-N",
        "input": "CCCOCCC",
        "text": "CCCOCCC"
    },
    "output": {
        "outcome": [
            0.36892343
        ]
    }
}
`

**Comment by luiscamachocaballero** [comment]
![Screenshot from 2023-10-23 11-21-05](https://github.com/ersilia-os/ersilia/assets/14283310/10cadf21-9ece-491d-bc3e-6c283a58f301)
![Screenshot from 2023-10-23 11-21-27](https://github.com/ersilia-os/ersilia/assets/14283310/130d4b65-7c52-4490-9957-933c7564ddbb)
![Screenshot from 2023-10-23 11-22-15](https://github.com/ersilia-os/ersilia/assets/14283310/38ba9f1d-81fd-4d0b-9833-77e74b5eccef)
![Screenshot from 2023-10-23 11-22-37](https://github.com/ersilia-os/ersilia/assets/14283310/23ba8813-4f51-465d-89ef-fad085fc741b)

Outputs of running Step 6. Identify Privileged Substructure for each molecule

Finally I could run the model after

**Comment by luiscamachocaballero** [comment]
Find one publication that describes an open source ML model that could be of interest to Ersilia (activity against a specific pathogen, cytotoxicity, side effects...) and link it in the thread.

**Comment by luiscamachocaballero** [comment]
prueba

**Comment by luiscamachocaballero** [comment]
My third choice is [**MRlogP: Transfer Learning Enables Accurate logP Prediction Using Small Experimental Training Datasets**](https://www.mdpi.com/2227-9717/9/11/2029/htm). This study on logP prediction using transfer learning is highly relevant due to its direct implications for drug discovery and medicinal chemistry. LogP, which measures the lipophilicity of a compound, is a critical property that influences its absorption, distribution, metabolism, and excretion (ADME). Accurate logP prediction is crucial for optimizing compound design and predicting its behavior in vivo. By improving logP prediction accuracy, this study offers a valuable tool for enhancing the drug discovery process, leading to the identification of more effective drug candidates. The study compares various logP prediction methods, highlighting their strengths and weaknesses. Understanding their performance characteristics enables researchers to select the most appropriate approach for their specific needs. Additionally, the introduction of transfer learning techniques in logP prediction showcases the potential for leveraging existing data to enhance modeling accuracy, even with limited experimental measurements.
Ultimately, this research has far-reaching implications beyond drug discovery. Accurate logP prediction is relevant in fields such as environmental chemistry, where understanding chemical behavior and fate is crucial; this study opens pathways for more accurate predictions in various scientific domains, contributing to advancements in both research and practical applications.

**Comment by luiscamachocaballero** [comment]
To run predictions, as required, I put the file [Essential Medicines List](https://raw.githubusercontent.com/ersilia-os/ersilia/master/notebooks/eml_canonical.csv) as input of the script IDL-PPBopt.ipynb, for compatibility, I have to edit the input's heading from smiles to cano_smiles.

This is the output
[IDL-PPBopt-output.csv](https://github.com/ersilia-os/ersilia/files/13169319/IDL-PPBopt-output.csv)

**Comment by luiscamachocaballero** [comment]
**Installing and running Docker**

I ran the code with `docker pull `ersiliaos/eos22io` to pull the IDL-PPB image from Ersilia Hub to docker, the model pulled successfully after ten minutes, it was downloaded almost 5 Gb data

**Comment by luiscamachocaballero** [comment]
My first selection is [**Predicting Antimalarial Activity in Natural Products Using Pretrained Bidirectional Encoder Representations from Transformers**](https://github.com/mldlproject/2021-NPBERT-Antimalaria) one publication that describes an open-source ML model that could be of interest to Ersilia. Predicting antimalarial activity in natural products is relevant because it can significantly expedite the process of discovering new potential drugs to combat malaria. Malaria is a life-threatening disease caused by the Plasmodium parasite and affects millions of people worldwide, particularly in **developing countries**. Traditional drug discovery methods can be time-consuming and costly. Therefore, the use of predictive models, such as machine learning, offers a more efficient approach.
By utilizing machine learning algorithms and incorporating chemical properties, researchers can develop models that predict the antimalarial activity of natural products. These models can save significant time and resources by narrowing down the pool of compounds that need to be experimentally tested for their antimalarial potential. Furthermore, natural products have long been recognized as important sources of drug leads, with many effective antimalarial drugs originating from natural sources. Predicting the antimalarial activity of natural products can identify promising candidates for further investigation, potentially leading to the discovery of novel antimalarial drugs.

About the software, there is an NPBERT pre-trained model with a configuration JSON file and a tokenizer. To run the model, you must follow the steps below: 

[Requirements:](https://github.com/mldlproject/2021-NPBERT-Antimalaria/tree/main/training/NPBERT_pretrained_model/save_model#requirements)

    rdkit == 2019.09.1.0
    transformer == 4.2.2
    python 3.7

**Generate features:**
[Generate feature with SMILES input](https://github.com/mldlproject/2021-NPBERT-Antimalaria/tree/main/training/NPBERT_pretrained_model/save_model#generate-feature-with-smiles-input)

python3 extract_feature.py --input_smile = "C1CCCCC1C2CCCCC2"
[Generate feature with *csv file containing SMILES inputs](https://github.com/mldlproject/2021-NPBERT-Antimalaria/tree/main/training/NPBERT_pretrained_model/save_model#generate-feature-with-csv-file-containing-smiles-inputs)
python3 extract_feature.py -- input_file= 'in_PATH/Intput.csv' --output_file = 'OUT_PATH/Output.csv'

The data is available but checkpoints aren't.

**Comment by GemmaTuron** [comment]
Hello,

Thanks for your work during the Outreachy contribution period, we hope you enjoyed it! We will now close this issue while we work on the selection of interns. Thanks again!
